### PR TITLE
Upload integration test logs on failure

### PR DESCRIPTION
**Pull request by tobim**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -34,16 +37,14 @@ style_task:
     - cmake --build build --target all
   unit_test_script:
     - cmake --build build --target test
-  integration_test_script:
-    - cmake --build build --target integration
+  integration_test_script: >
+      cmake --build build --target integration -- -d integration_test || (tar -czf $(cat ARTEFACT_NAME).tar.gz integration_test && scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i upload_key build/$(cat ARTEFACT_NAME).tar.gz mule@tenzir.dfn-cert.de:integration/vast/ && echo "Download with: `scp tenzir.dfn-cert.de:/home/mule/integration/vast/$(cat ARTEFACT_NAME).tar.gz .`" && false)
```
Comment (dominiklohmann) [inline .cirrus.yml]: I think this should be split up into multiple lines for readability and maintainability. Additionally, art**e**fact is the British spelling, and art**i**fact is the American spelling. This should be changed for consistency.

Content-wise: LGTM!

Comment (mavam) [reply .cirrus.yml]: Funny, I gave the exact same feedback regarding readability in the office earlier.

**Comment by tobim** [comment]
> Would it be possible to replace `tar -czf $(cat ARTEFACT_NAME).tar.gz` with `tar -czf $(echo $ARTEFACT_NAME).tar.gz`?

Environment variables are not preserved across multiple `_script` instructions, therefore I created the workaround via the files.

File: .cirrus.yml (hunk 1 above)
Comment (tobim) [reply .cirrus.yml]: > I think this should be split up into multiple lines for readability and maintainability. Additionally, art**e**fact is the British spelling, and art**i**fact is the American spelling. This should be changed for consistency.

Done and done.

**Pull request merged**